test/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+project(LittleTypeLibraryTests LANGUAGES CXX)
+
+# ---- Add root project ----
+
+# We want warnings in tests
+set(LittleTypeLibrary_INCLUDE_WITHOUT_SYSTEM ON CACHE INTERNAL "")
+
+include(FetchContent)
+FetchContent_Declare(LittleTypeLibrary SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/..")
+FetchContent_MakeAvailable(LittleTypeLibrary)
+
+# ---- Add googletest ----
+
+# Download only the first time
+set(FETCHCONTENT_UPDATES_DISCONNECTED ON)
+
+FetchContent_Declare(googletest
+        GIT_REPOSITORY "https://github.com/google/googletest.git"
+        GIT_TAG aee0f9d9b5b87796ee8a0ab26b7587ec30e8858e)
+FetchContent_MakeAvailable(googletest)
+
+# ---- Tests ----
+
+enable_testing()
+
+add_executable(LTLtest source/test.cpp)
+
+target_link_libraries(LTLtest PRIVATE
+        LittleTypeLibrary::LTL
+        gtest_main)
+
+target_compile_features(LTLtest PRIVATE cxx_std_17)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  target_compile_options(LTLtest PRIVATE -ftime-trace)
+endif()
```

Comment:
Regarding MSVC compile options, I added this :

```
if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  target_compile_options(LTLtest PRIVATE -ftime-trace)

elseif(MSVC)
    string(TOLOWER ${CMAKE_BUILD_TYPE} build_type)
    if(${build_type} STREQUAL "debug")
        set(BUILD_DEPENDENT_COMPILE_OPTION /MTd /Zi /DEBUG:FULL)
    else()
        set(BUILD_DEPENDENT_COMPILE_OPTION /MT)
    endif()

    target_compile_options(LTLtest PRIVATE
      /Zc:preprocessor
      /permissive-
      /d2cgsummary
      /Wv:18
      /bigobj
      ${BUILD_DEPENDENT_COMPILE_OPTION})
endif()
```

The experimental:preprocessor is necessary because of macro LTL uses.
MSVC standard macro are not standard macro. You must use experimental:preprocessor to LTL works well.

The /bigobj is necessary only for test because it is a big file. Just modify things for MSVC, check if my modification is well, and I am going to accept your PR.

Thanks for everything :).

Reply:
> The /bigobj is necessary only for test because it is a big file.

Looks like it's just gtest stuffing the object file with lots of junk.  
I can recommend doctest as a great alternative.